show/main.py

```diff
@@ -393,6 +395,44 @@ def run_command_in_alias_mode(command):
         sys.exit(rc)
 
 
+def get_bgp_summary_extended(command_output):
+    """
+    Adds Neighbor name to the show ip[v6] bgp summary command
+    :param command: command to get bgp summary
+    """
+    modified_output = []
+    my_list = iter(command_output.splitlines())
+    for element in my_list:
+        if element.startswith("Neighbor"):
+            element = "{}\tNeighborName".format(element)
+            modified_output.append(element)
+        elif not element or element.startswith("Total number "):
+            modified_output.append(element)
+        elif re.match(r"([0-9A-Fa-f]{1,4}:|\d+.\d+.\d+.\d+)", element.split()[0]):
+            ip = element.split()[0]
+            name = bgp_neighbor_ip_to_name(ip)
+            if len(element.split()) == 1:
+                modified_output.append(element)
+                element = next(my_list)
```

Comment:
I don't feel safe with next() here. I'm not sure it's a good practice.

Reply:
**your regexp doesn't have '*' prefix, which we have for dynamic neighbors.**
Dynamic neighbors are configured by providing subnets instead of defining neighbors explicitly.
But on both cases, ```show ip[v6] bgp summary``` output provides exact neighbor ip. So I have added regex to catch the ipv4 or v6 obtained from the command output.

**element = next(my_list)**
The ```show ipv6 bgp summary``` command output have ipv6 address in one line and the other datas in second line.
```
acb0::10:182:1:10e
                4 64802     548     546        0    0    0 08:59:22        7
```
as the motive of this PR is to add the Neighbor Name at the end of default data, we have to move to the line and append at the end. That is the reason I chose the next method.
Also, if something goes wrong here, we are catching it as an exception (**added in the same PR @ bgp_quagga and frr files**) and display the default output as today to the user.

I have addressed other comments as suggested. Please advise if the justification provided for the other functions are ok.